core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -660,12 +662,14 @@ private boolean hasBoundedCharacterType(Type type)
         protected Scope visitDelete(Delete node, Optional<Scope> scope)
         {
             Table table = node.getTable();
-            QualifiedObjectName tableName = createQualifiedObjectName(session, table, table.getName());
-            if (metadata.getView(session, tableName).isPresent()) {
+            QualifiedObjectName relationName = createQualifiedObjectName(session, table, table.getName());
```

Comment:
Maybe call this `originalName` to be clear that it's different than `tableName`, since `relationName` and `tableName` sound similar (not obvious that the wrong one is used)